### ISSUE TITLE
Compatibility with PyPy

### ISSUE THREAD

**Issue by smuuf** [Eric-Canas/CircularDict#2]
PyPy doesn't implement `sys.getsizeof(...)` and fails when it's used. Similarly to https://github.com/pandas-dev/pandas/issues/17228.

Reference:
> sys.getsizeof() always raises TypeError
https://doc.pypy.org/en/latest/cpython_differences.html

I'd would like to use this package, but I can't because of this - that's because even if I don't want to use the `maxsize_bytes` functionality,
`CircularDict` calls `sys.getsizeof(...)` in several places.

I think the solution could be to "hide" these calls behind a `is maxsize_bytes specified?` condition, for example something like this:

```diff
    def __setitem__(self, key: Any, value: Any):
        ...
-       item_size = sys.getsizeof(key) + sys.getsizeof(value)

        # If the element could not fit in the empty dictionary, raise an error
        if self.maxsize_bytes is not None:
+           item_size = sys.getsizeof(key) + sys.getsizeof(value)
            if item_size > self.maxsize_bytes:
                raise MemoryError(f"Item size {item_size} is larger than maxsize {self.maxsize_bytes}")

        ...
```

That way this package could be also used in PyPy - if one sacrifices the `maxsize_bytes` functionality when using PyPy.

What do you think? :)

**Comment by Eric-Canas** [comment]
Hi!

Sorry for the late response, I didn't notice the issue. 

I didn't know about that problem, and the solution you propose sounds actually good. 

I'll do it this way. 

Thanks!

**Comment by Eric-Canas** [comment]
Solved in [version 1.8](https://github.com/Eric-Canas/CircularDict/commit/55ee8afc7cb34554235b9e3fbd0db8defe0224f4). 

Already updated on [PyPi](https://pypi.org/project/circular-dict/1.8/)

Thanks for the suggestion! 

Please reopen it if you still find any incorrect behaviour.
